inventory-generation/identity-management/queue/user-management.yml

```diff
@@ -4,15 +4,7 @@
   include_vars:
     file: "{{ job.path }}"
 
-- name: Set fact with user information
-  set_fact:
-    user_info:
-      first_name: "{{ first_name | trim }}"
-      last_name: "{{ last_name | trim }}"
-      email: "{{ email | trim }}"
-      user_name: "{{ email.split('@')[0] | trim }}"
-      state: "absent"
-
 - name: "Add User to removal queue"
   set_fact:
-    users_remove: "{{ (users_remove | default([])) + [ user_info ] }}"
+    users_remove: "{{ users_remove|d([]) + [ item | combine({'state': 'absent'})] }}"
+  loop: "{{ users | selectattr('email', 'equalto', email ) | list }}"
```

Comment:
No sure I follow the logic here. Two thnings:
1) the `users` list isn't the correct one to use here
2) we still need the logic above to correctly generate the `user_name` based on e-mail.

... and 3) - which may have been an issue in the past as well - we should be able to process a list of users. IIRC the queue job file may include multiple users (I may remember incorrectly on this, but something to check), but if that's not the case we should be good as-is - just need to correct 1) and 2) above

Reply:
Remove Queue generation recreated (90874c5ee03226bc52eb48fc28663b75c1cb3659)